tests/test_session_affinity.py

```diff
@@ -501,6 +501,168 @@ def test_session_list_endpoint(self):
                 assert "last_used" in session_info
                 assert "alive" in session_info
 
+    def test_infinite_loop_timeout_then_simple_job(self, tester):
```

Comment:
can you please add a test that checks that state is restored. So something like 

first cell:
a = 2

second cell:
something that timeouts

third cell:
print(a + 4)

should work correctly as `a` from first cell should persist in a session

Reply:
they are separate from this test though because it's really two different units of functionality